Rule conflict between `no-invalid-aria-attributes` and `no-unnecessary-concat`

`no-invalid-aria-attributes` was recently patched to allow for handlebars as acceptable for aria attributes, #2389 e.g.: 
`<div aria-label="{{@name}}"></div>` 

However `no-unnecessary-concat` prefers `<div aria-label={{@name}}></div>` but `no-invalid-aria-attributes` marks this as invalid without the quotes. Both however correctly evaluate to a string.

It seems like `no-invalid-aria-attributes` should be aware of  `no-unnecessary-concat` requirements -- opening this issue per https://github.com/ember-template-lint/ember-template-lint/pull/2389#issuecomment-1053609716

@rrenno I'm not able to reproduce what you've encountered either. Can you retest using the latest patch/version of ember-template-lint? You may have used a code branch that was probably in an intermediary state of completion. Below are the results I'm encountering using `ember-template-lint@^4.3.0`.

`<div aria-label="{{@name}}"></div>`:
```sh
app/templates/application.hbs
  7:16  error  Unnecessary string concatenation. Use {{@name}} instead of "{{@name}}".  no-unnecessary-concat
✖ 1 problems (1 errors, 0 warnings)
```
`<div aria-label={{@name}}></div>`:
```sh
✨  Done in 1.14s.
```
Thanks!

cc @judithhinlung

In my effort to distill this down into a base case I think I over generalized the issue. 
Each incorrectly flagged instance contains a *conditional* -- which I believe should not be flagged by the linter, correct?.

E.g. 
`aria-label={{if @label @label @placeholder}}`
`aria-label={{unless @label @label @placeholder}}` 
```console
addon/components/zb-input/template.hbs
  16:4  error  The value for aria-label must be a string.  no-invalid-aria-attributes
  34:4  error  The value for aria-label must be a string.  no-invalid-aria-attributes
```

Switching to:
`aria-label="{{if @label @label @placeholder}}"`
`aria-label="{{unless @label @label @placeholder}}"` 
"Fixes" the `no-invalid-aria-attributes` error but then throws  
```console
addon/components/zb-input/template.hbs
  20:19  error  Unnecessary string concatenation. Use {{if @label @label @placeholder}} instead of "{{if @label @label @placeholder}}".  no-unnecessary-concat
  36:19  error  Unnecessary string concatenation. Use {{unless @label @label @placeholder}} instead of "{{unless @label @label @placeholder}}".  no-unnecessary-concat
```

I'm also noticing a similar issue with conditionals and `aria-describedby` when conditional without quotes is used. e.g.
```console
addon/components/zb-input/template.hbs
  33:4  error  The value for aria-describedby must be a list of strings that represent DOM element IDs (idlist)  no-invalid-aria-attributes
```

Sorry for the misdirection, let me know if I can help clarify further.

--- a/docs/rule/no-invalid-aria-attributes.md
+++ b/docs/rule/no-invalid-aria-attributes.md
@@ -11,7 +11,7 @@ This rule **forbids** the following:
 ```hbs
 <input type="text" aria-not-real="true" />
 <div role="region" aria-live="bogus">Inaccessible live region</div>',
-<button type="submit" aria-invalid={{if this.foo "true" "woosh"}}>Submit</button>
+<button type="submit" disabled="true" aria-disabled="123">Submit</button>
 ```
 
 This rule **allows** the following:

--- a/lib/rules/no-invalid-aria-attributes.js
+++ b/lib/rules/no-invalid-aria-attributes.js
@@ -84,23 +84,6 @@ function validityCheck(expectedType, permittedValues, allowUndefined, value) {
   }
 }
 
-function getValuesFromMustache(mustacheNode) {
-  let valuesList = [];
-  if (['BooleanLiteral', 'NumberLiteral', 'StringLiteral'].includes(mustacheNode.path.type)) {
-    valuesList.push(mustacheNode.path.original);
-  } else if (mustacheNode.path.type === 'PathExpression') {
-    if (mustacheNode.path.original === 'if' || mustacheNode.path.original === 'unless') {
-      if (mustacheNode.params.length === 2 || mustacheNode.params.length === 3) {
-        valuesList.push(mustacheNode.params[1].value);
-      }
-      if (mustacheNode.params.length === 3) {
-        valuesList.push(mustacheNode.params[2].value);
-      }
-    }
-  }
-  return valuesList;
-}
-
 export default class NoInvalidAriaAttributes extends Rule {
   visitor() {
     return {
@@ -126,24 +109,11 @@ export default class NoInvalidAriaAttributes extends Rule {
           let allowUndefined = validAriaAttribute.allowundefined || false;
           let isValidValue;
           // Skip validation for interpolated string values
-          if (attribute.value.type === 'ConcatStatement') {
+          if (
+            attribute.value.type === 'MustacheStatement' ||
+            attribute.value.type === 'ConcatStatement'
+          ) {
             return;
-          } else if (attribute.value.type === 'MustacheStatement') {
-            if (attribute.value.path) {
-              let valuesList = getValuesFromMustache(attribute.value);
-              if (valuesList.length === 0) {
-                isValidValue = true;
-              } else {
-                for (let value of valuesList) {
-                  isValidValue = validityCheck(
-                    expectedType,
-                    permittedValues,
-                    allowUndefined,
-                    value
-                  );
-                }
-              }
-            }
           } else {
             isValidValue = validityCheck(
               expectedType,

--- a/test/unit/rules/no-invalid-aria-attributes-test.js
+++ b/test/unit/rules/no-invalid-aria-attributes-test.js
@@ -22,6 +22,7 @@ generateRuleTests({
     '<button type="submit" aria-disabled={{this.isDisabled}}>Submit</button>',
     '<div role="textbox" aria-sort={{if this.hasCustomSort "other" "ascending"}}></div>',
     '<div role="combobox" aria-expanded="undefined"></div>',
+    '<button aria-label={{if @isNew (t "actions.add") (t "actions.edit")}}></button>',
   ],
   bad: [
     {
@@ -167,26 +168,6 @@ generateRuleTests({
       },
     },
     {
-      template: '<input type="password" required="true" aria-errormessage={{0}} />',
-      verifyResults(results) {
-        expect(results).toMatchInlineSnapshot(`
-          [
-            {
-              "column": 0,
-              "endColumn": 65,
-              "endLine": 1,
-              "filePath": "layout.hbs",
-              "line": 1,
-              "message": "The value for aria-errormessage must be a string that represents a DOM element ID",
-              "rule": "no-invalid-aria-attributes",
-              "severity": 2,
-              "source": "<input type=\\"password\\" required=\\"true\\" aria-errormessage={{0}} />",
-            },
-          ]
-        `);
-      },
-    },
-    {
       template:
         '<button type="submit" aria-describedby="blah false">Continue at your own risk</button>',
       verifyResults(results) {
@@ -202,26 +183,6 @@ generateRuleTests({
               "rule": "no-invalid-aria-attributes",
               "severity": 2,
               "source": "<button type=\\"submit\\" aria-describedby=\\"blah false\\">Continue at your own risk</button>",
-            },
-          ]
-        `);
-      },
-    },
-    {
-      template: '<button type="submit" aria-describedby={{false}} >broken button</button>',
-      verifyResults(results) {
-        expect(results).toMatchInlineSnapshot(`
-          [
-            {
-              "column": 0,
-              "endColumn": 72,
-              "endLine": 1,
-              "filePath": "layout.hbs",
-              "line": 1,
-              "message": "The value for aria-describedby must be a list of strings that represent DOM element IDs (idlist)",
-              "rule": "no-invalid-aria-attributes",
-              "severity": 2,
-              "source": "<button type=\\"submit\\" aria-describedby={{false}} >broken button</button>",
             },
           ]
         `);
@@ -262,26 +223,6 @@ generateRuleTests({
               "rule": "no-invalid-aria-attributes",
               "severity": 2,
               "source": "<div role=\\"heading\\" aria-level=\\"true\\">Another inaccessible heading</div>",
-            },
-          ]
-        `);
-      },
-    },
-    {
-      template: '<div role="heading" aria-level={{"blah"}}>Broken heading</div>',
-      verifyResults(results) {
-        expect(results).toMatchInlineSnapshot(`
-          [
-            {
-              "column": 0,
-              "endColumn": 62,
-              "endLine": 1,
-              "filePath": "layout.hbs",
-              "line": 1,
-              "message": "The value for aria-level must be an integer.",
-              "rule": "no-invalid-aria-attributes",
-              "severity": 2,
-              "source": "<div role=\\"heading\\" aria-level={{\\"blah\\"}}>Broken heading</div>",
             },
           ]
         `);
@@ -344,26 +285,6 @@ generateRuleTests({
               "rule": "no-invalid-aria-attributes",
               "severity": 2,
               "source": "<div role=\\"region\\" aria-live=\\"polite\\" aria-relevant=\\"additions errors\\">Inaccessible live region</div>",
-            },
-          ]
-        `);
-      },
-    },
-    {
-      template: '<input type="text" aria-required={{if this.foo "true" "woosh"}} />',
-      verifyResults(results) {
-        expect(results).toMatchInlineSnapshot(`
-          [
-            {
-              "column": 0,
-              "endColumn": 66,
-              "endLine": 1,
-              "filePath": "layout.hbs",
-              "line": 1,
-              "message": "The value for aria-required must be a boolean.",
-              "rule": "no-invalid-aria-attributes",
-              "severity": 2,
-              "source": "<input type=\\"text\\" aria-required={{if this.foo \\"true\\" \\"woosh\\"}} />",
             },
           ]
         `);

